src/section/mod.rs

```diff
@@ -164,6 +173,15 @@ impl Section {
         &self.chain
     }
 
+    // Extend the section chain so it starts at `new_first_key` while keeping the last key intact.
```

Comment:
it's actually: update the chain to be : from the new_first_key to the current_last_key within the full_chain, right?
I think the name of `extend` is bit improper, `slice` is more accurate ?
anyway, I am ok with stay with the current name, just thinking the comment can be more clear & accurate?

Reply:
I don't think `slice` is appropriate, but I think I can see the confusion. There are two chains here: first the chain we want to extend (`self.chain`) and second the full section chain we use to lookup the missing keys (`full_chain`). We are "extending" the first chain but "slicing" the second chain. However because it's the first chain that's the one being modified, the function name is `extend` which I think is appropriate.

Reply:
ok then, make sense.